### PR TITLE
Fixes #1603 - Adds test for tcms.core.utils.maito

### LINKED ISSUE #1603: Automated tests for missing coverage in tcms.core.utils.mailto
Kiwi TCMS needs additional automated test(s) as mentioned in subject.

**Definition of Done**:
- full coverage for the module or class mentioned in the title. This means both positive/negative scenarios where this makes sense or scenarios which traverse different branches of the code where that makes sense.
- test class/test method names are descriptive and reflect the behavior of the subject under test.  A good scheme to follow is `Test<ClassUnderTest>.test_[who][what][expected_results]`
- CI pass
- All code review passed & PR merged

Hints:
- Coverage report is at https://codecov.io/gh/kiwitcms/Kiwi/tree/master/tcms
- Devel getting started docs: https://kiwitcms.readthedocs.io/en/latest/contribution.html#testing
- Django testing documentation: https://docs.djangoproject.com/en/3.0/topics/testing/

*This issue is part of Kiwi TCMS open source bounty program. For more information see the link(s) in [bounty-program milestone](https://github.com/kiwitcms/Kiwi/milestone/4)*

### PR DESCRIPTION
**Pull request by pandafy**

### CODE REVIEW

File: tcms/core/tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import unittest
```
Comment (atodorov) [inline tcms/core/tests/test_utils.py:1]: See flake8 issues

File: tcms/core/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+class TestMailTo(unittest.TestCase):
+    @classmethod
+    def setUp(self):
+        self.subject = 'Test Subject'
+        self.body = 'Body text'
+        self.sender = 'kiwi@example.com'
+        self.recipients = None
+
+    @property
+    def test_kwargs(self):
```
Comment (atodorov) [inline tcms/core/tests/test_utils.py:20]: Rename to `expected_kwargs`

File: tcms/core/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                     self.sender,
+                     self.recipients),
+            'kwargs': {
+                'fail_silently': False
+            }
+        }
+
+    @patch('threading.Thread')
+    def test_string_recipient(self, mock):
+        self.recipients = ['example@example.com']
```
Comment (atodorov) [inline tcms/core/tests/test_utils.py:34]: Rename to `self.expected_recipients`

File: tcms/core/tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+               recipients='example@example.com',
+               context=self.body)
+        mock.assert_called_once_with(**self.test_kwargs)
+
+    @patch('threading.Thread')
+    def test_duplicate_recipients(self, mock):
+        self.recipients = ['example@example.com']
+        mailto(template_name=None,
+               subject=self.subject,
+               recipients=self.recipients * 4,
```
Comment (atodorov) [inline tcms/core/tests/test_utils.py:46]: hard-code the value here.

File: tcms/core/tests/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+from django.core.mail import send_mail
+from django.template.loader import render_to_string
+from unittest.mock import patch
+
+from tcms.core.utils.mailto import mailto
+
+
+class TestMailTo(unittest.TestCase):
+    @classmethod
+    def setUp(self):
```
Comment (atodorov) [inline tcms/core/tests/test_utils.py:13]: this is setUpClass(cls)

File: tcms/core/tests/test_utils.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        mock.assert_called_once_with(**self.test_kwargs)
+
+    @patch('threading.Thread')
+    def test_template(self, mock):
+        template_name = 'email/user_registered/notify_admins.txt'
+        context = {
+            'username': 'username',
+            'user_url': 'https://example.com/username/',
+        }
+        self.body = render_to_string(template_name, context)
```
Comment (atodorov) [inline tcms/core/tests/test_utils.py:79]: Rename to `self.expected_body`

File: tcms/core/tests/test_utils.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+    @patch('django.conf.settings.DEBUG', True)
+    @patch('django.conf.settings.ADMINS', [('Admin', 'admin@example.com')])
+    @patch('threading.Thread')
+    def test_admin_email_on_debug(self, mock):
+        self.recipients = ['example@example.com', 'admin@example.com']
+        mailto(template_name=None,
+               subject=self.subject,
+               recipients='example@example.com',
+               context=self.body)
```
Comment (atodorov) [inline tcms/core/tests/test_utils.py:69]: Hrd-code the value of the context argument to avoid using `self.expected_body`.

**Review by atodorov — CHANGES_REQUESTED** [review]
Coverage looks good, needs minor fixes and improvements for readability.

File: tcms/core/tests/test_utils.py (hunk 5 above)
Comment (pandafy) [reply tcms/core/tests/test_utils.py:13]: Initially I started with `setUpClass` method, but after writing a couple of tests I realised that we need to update different expected parameters according to tests. Changing this to `setUpClass` method will make those parameters update for subsequent subsequent tests. Having `setUp` method ensures that consistent initial condition is maintained for each test. 

I forgot to remove the `classmethod` decorator. I think using `setUp` method is the best thing to do here.

**Review by atodorov — APPROVED** [review]

**Pull request merged**